pyanaconda/modules/payloads/payload/dnf/dnf.py

```diff
@@ -54,11 +55,21 @@ def supported_source_types(self):
 
     def process_kickstart(self, data):
         """Process the kickstart data."""
-        pass
+        source_type = SourceFactory.get_rpm_type_for_kickstart(data)
+
+        if source_type is None:
+            return None
+
+        source = SourceFactory.create_source(source_type)
+
+        if source is not None:
```

Comment:
The method `create_source` will raise an exception if the source is unknown. Anyway, it is a bug if the factory doesn't know how to create a source returned by `get_rpm_type_for_kickstart`, so I don't think we need to handle it.

Reply:
ACK.